pennylane/gradients/spsa_gradient.py

```diff
@@ -54,9 +54,10 @@ def _rademacher_sampler(indices, num_params, *args, seed=None):
         by ``indices``, each entry sampled independently from the Rademacher distribution.
     """
     # pylint: disable=unused-argument
-    rng = np.random.default_rng(seed)
+    if seed is not None:
+        np.random.seed(seed)
     direction = np.zeros(num_params)
-    direction[indices] = rng.choice([-1, 1], size=len(indices))
+    direction[indices] = np.random.choice([-1, 1], size=len(indices))
```

Comment:
Why use `np.random` again?

Reply:
I couldn't figure out why tests were failing, see https://github.com/PennyLaneAI/pennylane/actions/runs/3892905264/jobs/6644875144

Tracking down why that test is failing would take more time than I planned on spending on this. Something to do with torch differentiation?